rustc_codegen_spirv/src/builder/intrinsics.rs

```diff
@@ -240,7 +241,33 @@ impl<'a, 'tcx> IntrinsicCallMethods<'tcx> for Builder<'a, 'tcx> {
                 if self.kernel_mode {
                     self.cl_op(CLOp::copysign, [args[0].immediate(), args[1].immediate()])
                 } else {
-                    self.fatal("TODO: Shader copysign not supported yet https://github.com/EmbarkStudios/rust-gpu/issues/148")
+                    let arg0 = args[0].immediate();
+                    let arg1 = args[1].immediate();
+                    let width = match self.lookup_type(arg0.ty) {
+                        SpirvType::Float(width) => width,
+                        other => panic!(
```

Comment:
Is panicking appropriate here? Shouldn't this produce a compile error?

Reply:
Typeck should already ensure that only floats get passed here. Using `unreachable!` or `bug!` makes sense here.

Reply:
Ah yeah, I guess I should have used the `bug!` macro - it's a bug/hard-panic, though, because it's breaking the invariants of what is known today (which is that the arguments to the `copysignf32` and `copysignf64` functions are floats). Similar with sizes that aren't 32 or 64 bits.

Reply:
https://github.com/rust-lang/rust/blob/20b1e05a8d0e1773dc840a3286aa37916e87d84b/compiler/rustc_typeck/src/check/intrinsic.rs#L251-L252